### ISSUE TITLE
Warning over camelCase when using variables from destructured objects

### ISSUE THREAD

**Issue by benhughes** [eslint/eslint#2254]
When destructuring objects in es6 you don't get warnings when using camel case, although if you attempt to use the variables you then receive warnings.

```javascript
// eslintrc rules
{"camelcase": [2, {"properties": "never"}]}

function checkObject(someObject) {
   // doesn't produce warnings 
   const {some_var, some_var2, some_var3} = someObject;
   // produces warnings as soon as you use the vars
   // Identifier 'some_var' is not in camel case. (camelcase)
   return some_var && some_var2 && some_var3;
}

```
Is there anyway to allow this use case?

**Comment by nzakas** [comment]
You would need to manually disable the rule. 

It's probably a bug that you're not getting a warning on the destructured assignment, since that is creating variables rather than properties.

**Comment by benhughes** [comment]
That's a shame.

I tried to do 

```javascript
const {some_var as someVar, some_var2 as someVar2} = someObject;
```
but it didn't like it. Then again if I'm doing doing that I may as well just refer to the object.

Thanks

**Comment by michaelficarra** [comment]
`as` should be `:`.
